### PR TITLE
feat(clickhouse): add ClickHouse as dependency

### PR DESCRIPTION
**Pull request by schnerring**
**Description**

Adds ClickHouse init and injector templates

⚒️ Fixes  # 

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

### CODE REVIEW

**Comment by schnerring** [comment]
Currently looking at the dependency injectors and I'm a little confused regarding the values:

```text
{{- $_ := set .Values.mariadb.url "plain"     ( ( printf "%v-%v" .Release.Name "mariadb" ) | quote ) }}
{{- $_ := set .Values.mariadb.url "plainhost" ( ( printf "%v-%v" .Release.Name "mariadb" ) | quote ) }}
{{- $_ := set .Values.mariadb.url "plainport"     ( ( printf "%v-%v:3306" .Release.Name "mariadb" ) | quote ) }}
{{- $_ := set .Values.mariadb.url "plainporthost" ( ( printf "%v-%v:3306" .Release.Name "mariadb" ) | quote ) }}
```

* `plain` == `plainhost`
* `plainport` == `plainporthost`

Are those kept for backwards compatibility, or what's the logic behind the naming convention?

Also, in the PG injector there seems to be a mismatch:

```text
plainporthost: {{ ( printf "%v-%v" .Release.Name "postgresql" ) | b64enc | quote }}
{{- $_ := set .Values.postgresql.url "plainporthost" ( ( printf "%v-%v:5432" .Release.Name "postgresql" ) | quote ) }}
```

I guess that's an error, just wanna make sure I don't misunderstand something.

**Comment by Ornias1993** [comment]
backwards compatibility only.

**Comment by schnerring** [comment]
Any ideas how to fix the unit test? I `CTRL+F`ed through the code but don't know what I'm missing.

**Comment by schnerring** [comment]
The failing test is unrelated to my changes. I reverted all the changes and only bumped the chart version and the test also failed. Sorry for the noise.

File: charts/common/templates/lib/dependencies/_clickhouseInit.tpl [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      drop:
+        - ALL
+  resources:
+  {{- with .Values.resources }}
+    {{- tpl ( toYaml . ) $ | nindent 4 }}
+  {{- end }}
+  command: [sh]
+  args:
+    - -c
+    - sleep 30
```
Comment (Ornias1993) [inline charts/common/templates/lib/dependencies/_clickhouseInit.tpl:21]: We still need an actuall init check here, this is going to cause a multitude of issues...

**Comment by stavros-k** [comment]
> The failing test is unrelated to my changes. I reverted all the changes and only bumped the chart version and the test also failed. Sorry for the noise.

Yep, it was there before :) np

**Comment by schnerring** [comment]
I merged the refactoring from the master branch, should be good to go. Not sure about the version bump: `10.3.0` or `10.2.1`?

File: charts/common/templates/lib/controller/_probes.tpl [hunk 2]
```diff
@@ -22,7 +22,7 @@ Probes selection logic.
             {{- if eq $probe.type "AUTO" -}}
               {{- $probeType = $primaryPort.protocol -}}
             {{- else -}}
-              {{- $probeType := $probe.type -}}
+              {{- $probeType = $probe.type -}}
```
Comment (schnerring) [inline charts/common/templates/lib/controller/_probes.tpl:25]: I'm pretty sure this is the reason [why I'm experiencing this issue with HTTP-type probes](https://github.com/truecharts/apps/pull/3060#issuecomment-1173814947)

**Comment by schnerring** [comment]
I merged main, should be good to go!

File: charts/common/templates/lib/controller/_probes.tpl (hunk 2 above)
Comment (schnerring) [reply charts/common/templates/lib/controller/_probes.tpl:25]: Nvm, I thought re-assigning with `:=` would ignore the re-assigned value. Doesn't seem to be an issue, though.

File: charts/common/templates/lib/dependencies/_clickhouseInjector.tpl [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  labels:
+    {{- include "tc.common.labels" . | nindent 4 }}
+  name: {{ $secretName }}
+stringData:
+  clickhouse-password: {{ $dbPass | quote }}
+  plainhost:           {{ $host | quote }}
+  plainporthost:       {{ $portHost | quote }}
+  ping:                {{ $ping | quote }}
+  url:                 {{ $url | quote }}
+  jdbc:                {{ $jdbc | quote }}
```
Comment (Ornias1993) [inline charts/common/templates/lib/dependencies/_clickhouseInjector.tpl:34]: With all the potential special characters I prefer the normal encoded secrets.

**Comment by Ornias1993** [comment]
I'm going to merge this, but the secret needs to be moved to encoded secret data instead of stringdata...

**Comment by Ornias1993** [comment]
That was fast @schnerring Thanks!

File: charts/common/templates/lib/controller/_probes.tpl [hunk 4]
```diff
@@ -7,7 +7,6 @@ Probes selection logic.
 {{- if $primaryService -}}
   {{- $primaryPort = get $primaryService.ports (include "tc.common.lib.util.service.ports.primary" (dict "serviceName" (include "tc.common.lib.util.service.primary" .) "values" $primaryService)) -}}
 {{- end -}}
-{{- $probeType := "TCP" -}}
```
Comment (Ornias1993) [inline charts/common/templates/lib/controller/_probes.tpl:10]: Beer remove the one down on line 13/14...

**Pull request merged**